healthcheck/tools/db/db.go

```diff
@@ -15,110 +15,65 @@
 package db
 
 import (
-	"errors"
-	"net/url"
+	"context"
 	"time"
 
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
-	"gopkg.in/mgo.v2"
-	"gopkg.in/mgo.v2/bson"
+	mgo "go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 var (
 	ErrMsgAuthFailedStr      string = "server returned error on SASL authentication step: Authentication failed."
 	ErrNoReachableServersStr string = "no reachable servers"
-	ErrSessionTimeout               = errors.New("timed out waiting for mongodb connection")
-	ErrPrimaryTimeout               = errors.New("timed out waiting for host to become primary")
 )
 
-func GetSession(cnf *Config) (*mgo.Session, error) {
-	cnf.DialInfo.Password = url.QueryEscape(cnf.DialInfo.Password)
-
-	if cnf.SSL == nil {
-		cnf.SSL = &SSLConfig{}
-	}
-
+func Dial(conf *Config) (*mgo.Client, error) {
 	log.WithFields(log.Fields{
-		"hosts":      cnf.DialInfo.Addrs,
-		"ssl":        cnf.SSL.Enabled,
-		"ssl_secure": !cnf.SSL.Insecure,
+		"hosts":      conf.Hosts,
+		"ssl":        conf.SSL.Enabled,
+		"ssl_secure": conf.SSL.Insecure,
 	}).Debug("Connecting to mongodb")
 
-	if cnf.DialInfo.Username != "" && cnf.DialInfo.Password != "" {
-		log.WithFields(log.Fields{
-			"user":   cnf.DialInfo.Username,
-			"source": cnf.DialInfo.Source,
-		}).Debug("Enabling authentication for session")
-	}
+	opts := options.Client().
+		SetHosts(conf.Hosts).
+		SetReplicaSet(conf.ReplSetName).
+		SetAuth(options.Credential{Password: conf.Password, Username: conf.Username}).
+		SetTLSConfig(conf.TLSConf).
+		SetServerSelectionTimeout(2 * time.Second).
+		SetConnectTimeout(3 * time.Second)
```

Comment:
@hors what do you think about these timeouts?

Reply:
@egegunes  it is not so simple to choose right timeouts. I think we need to ask experts about it. But for now we can avoid setting them and as soon as we have right numbers (consulting with experts can take some time) we will set them.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/51d5562b7016f66c33a6b43e707c68ecbe6d2390